### PR TITLE
Enable Binding Cluster for the all-clusters-app.zap configuration

### PR DESCRIPTION
**Pull request by vivien-apple**
This code is the result of enabling the `Binding Cluster` server & client into the `all-clusters-app` configuration file.

There is a lot of noise in the `.zap` configuration file. Sorry about that, we need to fix ZAP to avoid it.

Most of the code changes has been generated with `ZAP`. The list of manual actions are:
 * Run `./scripts/tools/zap_configure.sh` and enable `Binding Cluster` server & client on the first endpoint
 * Add `chip::ClusterId` to the list of readable types in `src/app/zap-templates/templates/app/helper.js` otherwise `call-command-handler.zapt` was failing to generate the proper output
 * Hand fix `examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h` since #4077 has not landed yet

### CODE REVIEW

File: src/controller/CHIPClusters.h [hunk 1]
```diff
@@ -72,6 +73,23 @@ class DLL_EXPORT BasicCluster : public ClusterBase
...
 };
 
+class DLL_EXPORT BindingCluster : public ClusterBase
+{
+public:
+    BindingCluster() : ClusterBase(kBindingClusterId) {}
+    ~BindingCluster() {}
+
+    // Cluster Commands
+    CHIP_ERROR Bind(Callback::Callback<> * onCompletion, uint64_t nodeId, uint16_t groupId, uint8_t endpointId,
```
Comment (andy31415) [inline src/controller/CHIPClusters.h:83]: should we use chip::endpointId fas a type for endpointid?

same for node and group, but unsure if we have typedefs for those (probably we should).

**Review by rwalker-apple — APPROVED** [review]
assuming type updates are made before merging

File: src/controller/CHIPClusters.h (hunk 1 above)
Comment (vivien-apple) [reply src/controller/CHIPClusters.h:83]: I have opened #4348 with a patch for this issue. It requires adding custom types to the ZCL definition inside `src/app/zap-templates/zcl`.

**Comment by vivien-apple** [comment]
> assuming type updates are made before merging

Done.

**Comment by vivien-apple** [comment]
I rebased this PR on top of #4077 now that it has landed. So the changes to `examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h` are generated as well.

**Review by jelderton — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**